articles/ros2_threat_model.md

```diff
@@ -0,0 +1,3291 @@
+secure enclave (TPM).</li>
+        <li>Robot filesystem should be wiped out if the robot perimeter is
+breached.</li>
+      </ul>
+    </td>
+    <td> </td>
+  </tr>
+
+  <tr>
+    <td>An attacker saturates the robot disk with data.</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td class="success">1</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td>13</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td>
+      <ul>
+        <li>Robot components disk quota should be bounded.</li>
+        <li>Disk usage should be properly monitored, logged and reported.</li>
+        <li>Optionally, components may have the option to run w/o any file
+system access. This should be prefered whenever possible.</li>
+      </ul>
+    </td>
+    <td> </td>
+  </tr>
+
+  <tr><th colspan="29">Embedded / Software / Logs</th></tr>
+
+  <tr>
+    <td>An attacker exfiltrates log data to a remote server.</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="warning">2</td>
+    <td class="warning">2</td>
+    <td class="warning">2</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td>12</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td>
+      <ul>
+        <li>Logs should never contain private data. Log data should be
+anonymized when needed.</li>
+        <li>Logs should be rotated and deleted after a pre-determined retention
+period.</li>
+        <li>Logs should be encrypted in-transit and at-rest.</li>
+        <li>Logs access should be ACL protected.</li>
+        <li>Logs access should be monitored to enable later audits.</li>
+      </ul>
+    </td>
+    <td> </td>
+  </tr>
+
+  <tr><th colspan="29">Embedded / Hardware / Sensors</th></tr>
+
+  <tr>
+    <td>An attacker spoofs a robot sensor (by e.g. replacing the sensor itself
```

Comment:
True, I agree with this threat evaluation overall but I'm missing in the "Embedded / Hardware / <Sensor/Actuators>" section considerations of physical attacks that involve complete removal (not simple replacement) of the sensor. I still find shocking how several commercial and industrial robots don't provide any mechanism to secure hardware physically and simply present sensors totally exposed. I collaborated in the production of a framework (with a slightly different aim) that touched into these aspects https://arxiv.org/pdf/1806.04042.pdf.

Would you like me to make a few additional row suggestions? @olaldiko was also involved on this so maybe he'd like to jump in directly?

Reply:
This would be in line with adding physical security measures or tamper detections. In would combine them into spoofing/removal of the sensor, and add alerting/logging of physical intrusions and sensor communication losses as mitigation.

Reply:
I agree this is a real concern. @vmayoral yes, adding a few rows in the doc would be great.